### PR TITLE
Decouple PluginFlags

### PR DESCRIPTION
**Pull request by mxswat**

### CODE REVIEW

**Comment by TanninOne** [comment]
I don't think this is complete. It still imports from gamesupport.ts which imports from "path" and "fs" so that dependency is still there.

File: src/views/PluginFlags.tsx [hunk 1]
```diff
@@ -93,11 +27,18 @@ function warningText(t: TranslationFunction, key: string) {
...
+    gameMode,
+    t,
+    doesSupportsESL,
+    getMinRevision,
+    isGameSupported,
+  } = props;
 
   const flags: JSX.Element[] = [];
 
-  if (!gameSupported(gameMode)) {
```
Comment (TanninOne) [inline src/views/PluginFlags.tsx:100]: As discussed yesterday I would like to avoid unnecessary changes.
Renaming gameSupported -> isGameSupported, supportsESL -> doesSupportsESL, ... is purely preference. If everyone did this we'd just go around in circles renaming each others stuff

**Review by TanninOne — CHANGES_REQUESTED** [review]

File: src/views/PluginFlags.tsx [hunk 2]
```diff
@@ -1,90 +1,24 @@
...
 import * as React from 'react';
 
 type TranslationFunction = typeof I18next.t;
 
 interface IBaseProps {
   plugin: IPluginCombined;
   gameMode: string;
+  isGameSupported: (gameMode: string) => boolean;
+  getMinRevision: (gameMode: string) => number;
+  doesSupportsESL: (gameMode: string) => boolean;
```
[the remark this answers is not in the document]
Comment (mxswat) [reply src/views/PluginFlags.tsx]: Updated

File: src/util/getPluginFlags.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable max-lines-per-function */
```
[the remark this answers is not in the document]
Comment (mxswat) [reply src/util/getPluginFlags.ts:1]: The function is still too long to not trigger the ES rules.

File: src/util/getPluginFlags.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import { IPluginCombined } from '../types/IPlugins';
+import I18next from 'i18next';
+import { gameSupported, supportsESL, minRevision } from '../util/gameSupport';
+type TranslationFunction = typeof I18next.t;
+
+export function getPluginFlags(plugin: IPluginCombined,
+                               t: TranslationFunction,
+                               gameMode: string): string[] {
```
[the remark this answers is not in the document]
Comment (mxswat) [reply src/util/getPluginFlags.ts]: Updated

File: src/views/PluginFlags.tsx (hunk 1 above)
Comment (mxswat) [reply src/views/PluginFlags.tsx]: Renaming reverted

File: src/views/PluginList.tsx [hunk 5]
```diff
@@ -1173,8 +1201,15 @@ class PluginList extends ComponentEx<IProps, IComponentState> {
         edit: {},
         isSortable: true,
         customRenderer: (plugin: IPluginCombined, detail: boolean, t: TranslationFunction) =>
-          (<PluginFlags plugin={plugin} gameMode={this.props.gameMode} t={t} />),
-        calc: (plugin: IPluginCombined, t) => getPluginFlags(plugin, t, this.props.gameMode),
+          (<PluginFlags
+            gameSupported={gameSupported}
```
[the remark this answers is not in the document]
Comment (mxswat) [reply src/views/PluginList.tsx]: Done

File: src/views/PluginList.tsx [hunk 6]
```diff
@@ -1097,8 +1103,18 @@ class PluginList extends ComponentEx<IProps, IComponentState> {
       minRevision,
       supportsESL,
       getPluginFlags,
+      gameMode,
     } = this.props;
 
+    const calcFlags = (plugin: IPluginCombined, t) =>
```
[the remark this answers is not in the document]
Comment (mxswat) [reply src/views/PluginList.tsx]: Fixed

File: src/views/PluginFlags.tsx [hunk 7]
```diff
@@ -5,29 +5,27 @@ import { tooltip } from 'vortex-api';
...
-                               gameId: string): string[] {
+export function getPluginFlags(t: TranslationFunction,
+                               plugin: IPluginCombined,
+                               gameSupported: boolean,
+                               supportsESL: boolean,
+                               minRevision: number): string[] {
   const result: string[] = [];
 
-  if (!gameSupported(gameId)) {
+  if (gameSupported) {
```
[the remark this answers is not in the document]
Comment (mxswat) [reply src/views/PluginFlags.tsx]: Fixed
